src/main/java/com/interview/coding/tasks/codereview/cache/UserService.java

```diff
@@ -0,0 +1,29 @@
+package com.interview.coding.tasks.codereview.cache;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/**
+ * Need to perform a code review of this cache solution.
+ * Then need to fix all found issues.
+ *
+ * @author Aliaksei Tatarynchyk
+ */
+@Service
+public class UserService {
+    static Map cache = new HashMap<>();
```

Comment:
It would be better to use a well-known cache solution then write your own, you can use e.g. Ehcache.